src/Traits/MessageResponseTrait.php

```diff
@@ -29,12 +29,11 @@ public function getProtocolVersion(): string
         return $this->response->getProtocolVersion();
     }
 
-    /**
-     * @return MessageInterface
-     */
     public function withProtocolVersion($version): MessageInterface
     {
-        return $this->response->withProtocolVersion($version);
+        $this->response->withProtocolVersion($version);
```

Comment:
We need to return the `withProtocolVersion($version)` otherwise we'll lost the response. We cannot return $this, the [PSR-7](https://www.php-fig.org/psr/psr-7/) MessageInterface is immutable.

Reply:
In that case you need to do a bigger refactoring in order to resolve this properly. withProtocolVersion() should not return a different implementation of MessageInterface

Reply:
maybe a good solution would be to clone the object and swap the response object with the new one and then return that.

Reply:
I think this is more a PHPStan issue rather that a real PHP one. In the meantime we'll leave the extra comment to prevent the error message from PHPStan.